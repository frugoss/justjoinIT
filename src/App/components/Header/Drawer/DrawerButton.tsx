import React from 'react';
import {Link} from 'react-router-dom';
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles,createStyles } from '@material-ui/core/styles';
import styles from './drawer.module.scss'


type DrawerButtonProps = {
    to: string,
    text: string,
    img: React.ReactNode,
    setOpen:  React.Dispatch<React.SetStateAction<boolean>>,
    logout?: () => void
}
const useStyles = makeStyles(createStyles({
    root: {
        color: "#B0BAC9",
        fontSize: 14,
        textDecoration: "none",
        fontWeight: 400,
        lineHeight: 1.5,
        width: "100%",
        alineHeight: 56,
        padding: "6px 6px 5px 0px",
    },
    listItem: {
        width: 310
    },
    listIcon: {
        marginLeft: 5
    }
}));

const DrawerButton: React.FC<DrawerButtonProps> = ({to, text, img, setOpen, logout}) => {
    const classes = useStyles();
    return (
        <>
            <Link  onClick={() => setOpen(false)} to={to} className={styles.linkStyle}>
                <ListItem onClick={to === "/" ? () => logout && logout() : () => ""} className={classes.listItem}>
                    <ListItemIcon className={classes.listIcon}>
                        {img}
                    </ListItemIcon>
                    <ListItemText  className={classes.root} primary={text} />
                </ListItem>
            </Link>
        </>
    );
};


export default DrawerButton
