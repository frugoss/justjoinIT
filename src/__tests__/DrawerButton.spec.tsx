import DrawerButton, {DrawerButtonProps} from "../App/components/Header/Drawer/DrawerButton";
import {render} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import React from 'react'
import {BrowserRouter} from "react-router-dom";


function renderDrawerButton(props: Partial<DrawerButtonProps> = {}) {
    const defaultProps: DrawerButtonProps = {
        to: "to",
        img: "img",
        text: "text",
        setOpen(){
        return},
        logout(){
        }
    }

    return render(<BrowserRouter><DrawerButton {...defaultProps} {...props}/> </BrowserRouter>);
};

describe("<DrawerButton />", () => {
    test("drawer button snapshot", () => {
        const {container} = renderDrawerButton();
        expect(container).toMatchSnapshot();
        }
    )
})
