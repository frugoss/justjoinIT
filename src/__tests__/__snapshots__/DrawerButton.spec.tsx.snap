// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerButton /> drawer button snapshot 1`] = `
<div>
  <a
    class="linkStyle"
    href="/to"
  >
    <li
      class="MuiListItem-root makeStyles-listItem-2 MuiListItem-gutters"
    >
      <div
        class="MuiListItemIcon-root makeStyles-listIcon-3"
      >
        img
      </div>
      <div
        class="MuiListItemText-root makeStyles-root-1"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          text
        </span>
      </div>
    </li>
  </a>
   
</div>
`;
